src/routes/jira/server/jira-ghe-servers.ts

```diff
@@ -0,0 +1,24 @@
+import { NextFunction, Request, Response } from "express";
+import { GitHubServerApp } from "models/github-server-app";
+import { groupBy, chain } from "lodash";
+
+export const JiraGheServers = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+): Promise<void> => {
+	try {
+		req.log.debug("Received Jira GHE servers page request");
+
+		const allServers = await GitHubServerApp.findForInstallationId(res.locals.installation.id) || [];
+		const gheServers = chain(groupBy(allServers, "gitHubBaseUrl")).map((_, key) => ({ displayName: key })).value();
```

Comment:
Theres a lot going on. I would consider simplifying this, even if its extra code.

Are you using the index as the displayName intentionally?

Reply:
Firstly, I am grouping the list of GHE servers based on their `gitHubBaseUrls`, since there could be multiple GHE server apps with same gitHubBaseUrl for a single instance. Then doing a map to just fetch the urls. Are you referring to break this into multiple lines?
Yes, I am mapping the index to `displayName` intentionally because the UI component that will receive this data is reusable component. It will be later used again for similar purpose. So, I picked `displayName` as a general index, to make it easy to understand.

Reply:
@joshkay10 it was already complicated before he started. I went through it with him and simplifying this function would introduce more risk than needed.

We definitely need to simplify, but I think should be done as a separate PR after we release GHE support